app/lib/redis.server.ts

```diff
@@ -0,0 +1,28 @@
+import Redis from 'ioredis';
+
+import type { Redis as RedisType, RedisOptions } from 'ioredis';
+
+let redis: RedisType;
+
+declare global {
+  var __redis: RedisType | undefined;
+}
+
+const redisOptions: RedisOptions = {
+  maxRetriesPerRequest: null,
+  enableReadyCheck: false,
+};
+
+// this is needed because in development we don't want to restart
+// the server with every change, but we want to make sure we don't
+// create a new connection to the Redis with every change either.
+if (process.env.NODE_ENV === 'production') {
+  redis = new Redis(process.env.REDIS_URL || '', redisOptions);
```

Comment:
Will we have a dedicated redis that survives between releases?

Reply:
We can run our own instance on the cluster, yes.  So when we re-deploy, it will remain.

Reply:
Perfect!